Course/Course/Presentation/Unit/CourseUnitView.swift

```diff
@@ -384,6 +145,294 @@ public struct CourseUnitView: View {
+            }
+        }
+    }
+    // swiftlint:enable function_body_length
+
+    private func viewOffset(for index: Int, with size: CGSize, insets: EdgeInsets) -> CGPoint {
+        let rightInset = (isHorizontal ? insets.trailing * CGFloat(index) : 0)
+        let x: CGFloat = isHorizontalNavigation ? -(size.width * CGFloat(index) + rightInset) : 0
+        let y: CGFloat = isHorizontalNavigation ? 0 : -(size.height * CGFloat(index))
+        return CGPoint(x: x, y: y)
+    }
+    
+    private func dropdown(block: CourseBlock) -> some View {
+        HStack {
+            if block.type == .video {
+                let title = block.displayName
+                Text(title)
+                    .lineLimit(1)
+                    .font(Theme.Fonts.titleLarge)
+                    .foregroundStyle(Theme.Colors.textPrimary)
+                    .padding(.vertical, 10)
+                    .padding(.horizontal, 20)
+                Spacer()
+            }
+        }
+    }
+
+    // MARK: - Course Navigation
+
+    private var courseNavigation: some View {
+        VStack(spacing: 0) {
+            ZStack {
+                if !isDropdownActive {
+                    GeometryReader { reader in
+                        VStack {
+                            HStack {
+                                let currentBlock = viewModel.verticals[viewModel.verticalIndex]
+                                    .childs[viewModel.index]
+                                if currentBlock.type == .video {
+                                    let title = currentBlock.displayName
+                                    Text(title)
+                                        .lineLimit(1)
+                                        .font(Theme.Fonts.titleLarge)
+                                        .foregroundStyle(Theme.Colors.textPrimary)
+                                        .padding(.leading, isHorizontal ? 30 : 42)
+                                        .padding(.top, isHorizontal ? 14 : 2)
+                                    Spacer()
+                                }
+                            }
+                            .frame(maxWidth: isHorizontal ? reader.size.width * 0.5 : nil)
+                            Spacer()
+                        }
+                    }
+                }
+                navigationBar
+                courseNavigationView
+            }
+        }
+        .frame(maxWidth: .infinity)
+    }
+
+    private var navigationBar: some View {
+        VStack(spacing: 0) {
+            ZStack(alignment: .bottom) {
+                NavigationBar(
+                    title: isDropdownActive ? sequenceTitle : "",
+                    leftButtonAction: {
+                        if viewModel.blockChanged {
+                            NotificationCenter.default.post(
+                                name: .onRefreshCourse,
+                                object: nil
+                            )
+                        }
```

Comment:
@eyatsenkoperpetio  In which case this notification gets posted, I've checked this isn't calling when setting the `blockChanged = true`?